azurerm/resource_arm_eventgrid_topic_test.go

```diff
@@ -0,0 +1,142 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAzureRMEventGridTopic_basic(t *testing.T) {
+	ri := acctest.RandInt()
+	config := testAccAzureRMEventGridTopic_basic(ri)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMEventGridTopicDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMEventGridTopicExists("azurerm_eventgrid_topic.test"),
```

Comment:
It would be nice to add more checks here, at least basic state checks with `resource.TestCheckResourceAttr` and/or `resource.TestCheckResourceAttrSet`.